client/src/components/TopBar.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from "react";
+import "./TopBar.css";
+
+// Validation utility function
+const isValidInput = (title, url) => {
```

Comment:
👍 Good UX that you're validating first in the browser, the user will get feedback faster than if they had to wait for a request to come back.